packages/functions/src/index.ts

```diff
@@ -3,8 +3,53 @@ import * as functions from "firebase-functions";
 import {checkAuthToken} from "./middlewares/check-auth-token";
 import authRouter from "./resources/auth/auth.router";
 import userRouter from "./resources/user/user.router";
+import * as admin from "firebase-admin";
+import {Change, onDocumentCreated,
+  onDocumentDeleted,
+  onDocumentUpdated,
+  QueryDocumentSnapshot} from "firebase-functions/v2/firestore";
+import {updatePublishedEvent}
+  from "./resources/bot/actions/updatePublishedEvent";
+import {sendToChannel} from "./resources/bot/actions/sendToChannel";
+import {deleteTelegramMessage}
+  from "./resources/bot/actions/deleteTelegramMessage";
+
+admin.initializeApp();
+
+export const publishToTelegram =
+onDocumentCreated("events/{eventId}", (event) => {
+  const snapshot = event.data;
+  if (snapshot) {
+    const data = snapshot.data();
+    sendToChannel(data);
+  }
+});
+
+
+export const deleteEventFromTelegram =
+ onDocumentDeleted("events/{eventId}", (event) => {
+   const snapshot = event.data;
+   if (snapshot) {
+     const data = snapshot.data();
+     deleteTelegramMessage(data);
+   }
+ });
+
+export const updatePublishedEventTrigger =
+ onDocumentUpdated("events/{eventId}", (event) => {
+   const change = event.data as Change<QueryDocumentSnapshot>;
+
+   const beforeData = change.before.data() as admin.firestore.DocumentData;
+   const afterData = change.after.data() as admin.firestore.DocumentData;
+
+   if (JSON.stringify(beforeData) !== JSON.stringify(afterData)) {
```

Comment:
not every field update should trigger event republishing, it's better to write function that compares fields that are visible in the channel. And also if status before was 'Draft' and become 'Active', sendToChannel should be called instead of updatePublishedEvent